### ISSUE TITLE
Docstring in a where clause is removed

### ISSUE THREAD

**Issue by Profpatsch** [tweag/ormolu#715]
**Describe the bug**

The following snippet

```
foo = ()
  where
     -- | foo
     x = ()
```

is changed by ormolu to

```
foo = ()
  where
    x = ()
```

that is the comment is deleted.

If I use a normal comment like

```
foo = ()
  where
    -- foo
    x = ()
```

that comment is preserved.

I ran into this, because I have a habit of documenting where-clauses with normal docstrings, which means promoting a where clause into a toplevel function is just a dedent away.

**Environment**
 - OS name + version: nixos 21.05
 - Version of the code: 
 ```
 > ormolu --version
ormolu 0.1.4.1 UNKNOWN UNKNOWN
using ghc-lib-parser 8.10.4.20210206
```

**Comment by mheinzel** [comment]
I think this is a known limitation of the GHC parser: https://github.com/tweag/ormolu/issues/474
It seems like it will be fixed by switching to ghc-lib-parser-9.0.1 in the future.

**Comment by mrkkrp** [comment]
Yeah, this is a duplicate of #474. It should be fixed as part of 2021 Ormolu internship.

**Comment by Profpatsch** [comment]
Great, just saw the changelog! Does that mean it’s fixed when I use ormolu 0.2.x.x with GHC 9.x.x?

**Comment by mrkkrp** [comment]
@Profpatsch I'll probably be fixed when we switch to 9.2 see https://github.com/tweag/ormolu/issues/474#issuecomment-889182337.
